docs/userguide/app_caching.rst

```diff
@@ -3,56 +3,60 @@
 App caching
 ----------
 
-When developing a workflow, developers often run the same workflow
-with incremental changes over and over. Often large fragments of
-a workflow will not have changed, yet apps will be re-executed, wasting
-valuable developer time and computation resources. App caching
-solves this problem by storing results from apps that have completed
-so that they can be re-used. App caching can be enabled by setting the ``cache``
-argument in the :func:`~parsl.app.app.python_app` or :func:`~parsl.app.app.bash_app` decorator to ``True`` (by default it is ``False``). App caching
-can be globally disabled by setting ``app_cache=False``
-in the :class:`~parsl.config.Config`.
+There are many situations in which a program may be re-executed
+over time. Often large fragments of the program will not have changed, 
+and therefore, re-execution of apps will waste valuable time and 
+computation resources. Parsl's app caching solves this problem by 
+storing results from apps that have successfully completed
+so that they can be re-used. 
+
+App caching is enabled by setting the ``cache``
+argument in the :func:`~parsl.app.app.python_app` or :func:`~parsl.app.app.bash_app` 
+decorator to ``True`` (by default it is ``False``). 
 
 .. code-block:: python
 
    @bash_app(cache=True)
    def hello (msg, stdout=None):
        return 'echo {}'.format(msg)
+			
+App caching can be globally disabled by setting ``app_cache=False``
+in the :class:`~parsl.config.Config`.
 
-
-App caching can be particularly useful when developing interactive workflows such as when
+App caching can be particularly useful when developing interactive programs such as when
 using a Jupyter notebook. In this case, cells containing apps are often re-executed
 during development. Using app caching will ensure that only modified apps are re-executed.
 
 
-Inputs
-^^^^^^
+App equivalence 
+^^^^^^^^^^^^^^^
 
 Two app invocations are treated as the same by the caching mechanism if their
```

Comment:
'treated as the same' doesn't seem quite right, but I'm not sure what would be better. 'determined to be equivalent' maybe?